src/commonMain/kotlin/org/kosat/heuristics/Selector.kt

```diff
@@ -0,0 +1,63 @@
+package org.kosat.heuristics
+
+import org.kosat.CDCL
+import org.kosat.Clause
+import org.kosat.Lit
+import kotlin.math.abs
+
+abstract class Selector {
+    protected var assumptions: List<Lit> = emptyList()
+
+    fun initAssumptions(assumptions: List<Lit>) {
+        this.assumptions = assumptions
+    }
+
+    abstract fun build(clauses: List<Clause>)
+    abstract fun nextDecisionVariable(vars: List<CDCL.VarState>, level: Int): Int
+    abstract fun addVariable()
+    abstract fun update(lemma: Clause)
+}
+
+class VSIDS(private var varsNumber: Int = 0) : Selector() {
+    private val decay = 50
+    private val divisionCoeff = 2.0
+    private var numberOfConflicts = 0
```

Comment:
This property should be in Solver (by Solver I mean CDCL class)... However, this class is not coupled with a Solver, similar to other "extracted classes"... Care to elaborate, why?